contracts/src/v0.8/dev/LyraAutomator.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import "../interfaces/AutomationCompatibleInterface.sol";
+import "../ConfirmedOwner.sol";
+import "./OptionMarketInterface.sol";
+
+contract LyraAutomator is AutomationCompatibleInterface, ConfirmedOwner {
+  OptionMarketInterface public s_optionMarket;
+
+  event BoardExpired(uint256 indexed boardId, uint256 blocknumber);
+  event SettlementFailed(uint256 indexed boardId, bytes lowLevelData);
+
+  constructor(OptionMarketInterface optionMarket) ConfirmedOwner(msg.sender) {
+    s_optionMarket = optionMarket;
+  }
+
+  function setOptionMarket(OptionMarketInterface optionMarket) external onlyOwner {
+    s_optionMarket = optionMarket;
+  }
+
+  function checkUpkeep(bytes calldata checkData)
+    external
+    override
+    returns (bool upkeepNeeded, bytes memory performData)
+  {
+    uint256[] memory liveBoards = s_optionMarket.getLiveBoards();
+    uint256 index = 0;
+
+    for (uint256 i = 0; i < liveBoards.length; i++) {
+      uint256 boardId = liveBoards[i];
+      OptionBoard memory board = s_optionMarket.getOptionBoard(boardId);
+      if (board.expiry < block.timestamp) {
+        liveBoards[index++] = boardId;
```

Comment:
interesting, you're manipulating the array itself. Confusing but would work

Reply:
Lyra